crates/cxx-qt-build/src/dependencies.rs

```diff
@@ -0,0 +1,301 @@
+    /// [Self::export_include_directory] function.
+    pub fn export_include_prefixes<'a>(
+        mut self,
+        prefixes: impl IntoIterator<Item = &'a str>,
+    ) -> Self {
+        let prefixes = prefixes.into_iter().map(|item| item.to_string()).collect();
+
+        let mut exported_prefixes = self
+            .exported_include_directories
+            .iter()
+            .map(|(_path, prefix)| prefix);
+        for prefix in &prefixes {
+            if let Some(duplicate) =
+                exported_prefixes.find(|exported_prefix| exported_prefix.starts_with(prefix))
+            {
+                panic!("Duplicate export_prefix! Cannot export `{prefix}`, as `{duplicate}` is already exported as an export_include_directory!");
+            }
+        }
+
+        self.exported_include_prefixes = prefixes;
+        self
+    }
+
+    /// Add a directory that will be added as an include directory under the given prefix.
+    ///
+    /// The prefix will automatically be exported (see also: [Self::export_include_prefixes])
+    ///
+    /// This function will panic if the given prefix is already exported.
+    pub fn export_include_directory(mut self, directory: impl AsRef<Path>, prefix: &str) -> Self {
+        let mut exported_prefixes = self.exported_include_prefixes.iter().chain(
+            self.exported_include_directories
+                .iter()
+                .map(|(_path, prefix)| prefix),
+        );
+        if let Some(duplicate) =
+            exported_prefixes.find(|exported_prefix| exported_prefix.starts_with(prefix))
+        {
+            panic!("Duplicate export_prefix! Cannot export `{prefix}`, as `{duplicate}` is already exported!");
+        }
+
+        self.exported_include_directories
+            .push((directory.as_ref().into(), prefix.to_owned()));
+        self
+    }
+
+    /// Reexport the dependency with the given link name.
+    /// This will make the dependency available to downstream dependencies.
+    ///
+    /// Specifically it will reexport all include_prefixes of the given dependency
+    /// as well as any definitions made by that dependency.
+    ///
+    /// Note that the link name may differ from the crate name.
+    /// Check your dependencies manifest file for the correct link name.
+    pub fn reexport_dependency(mut self, link_name: &str) -> Self {
+        self.reexport_links.insert(link_name.to_owned());
+        self
+    }
+}
+
+// cxx-qt-lib example
+// Interface {
+//     export_crate_headers: false,
+//     exported_header_prefixes: Some(vec!["cxx-qt-lib/", "python3"]),
+//     generated_path: "cxx-qt-lib-internals",
+// }
+
+// cxx-qt-lib-headers example
+// Interface {
+//     export_crate_headers: true,
+//     exported_header_prefixes: Some(vec!["cxx-qt-lib-extras/..."]),
+//     reexport_links: vec!["cxx-qt-lib"],
+// }
+
+#[derive(Clone, Serialize, Deserialize)]
+// This struct is used by cxx-qt-build internally to propagate data through to downstream
```

Comment:
could be a proper doc comment? then it would appear when you document locally with the private things too?

Reply:
done